app/controllers/time_slot_controller.rb

```diff
@@ -38,10 +38,13 @@ def schedule
   end
 
   def cancel
-    @appointment = Appointment.find(cancel_params[:appointment_id])
-    @appointment.update(patient_id: nil)
+    Appointment.transaction do
+      @current_patient.appointments
+        .where(id: cancel_params[:appointment_id])
+        .update_all(patient_id: nil)
```

Comment:
```suggestion
        .find(id: cancel_params[:appointment_id])
        .update(patient_id: nil)
```

Que tal deixar assim mesmo para otimizar a query?

Reply:
Então, apesar de ficar mais idiomático, isso faria duas queries, uma pro `find` e outra pro `update`. Fiz com o `update_all` porque assim vira de fato uma instrução `UPDATE`. Só preciso verificar a questão do `@current_patient.appointments.where`, mas pelo que eu lembre ele também já vai restringir a busca aos agendamentos do paciente.

Reply:
Boa!!! Vi nos logs aqui que o `find` + `update` faz mais buscas no banco.

Reply:
Só ainda não sei qual vai ser mais vantajoso, usando o `find` + `update` adiciona mais query, mas são limitadas a 1 registro. Também tem a questão de ficar mais legível

Reply:
Como o `where` tá buscando pela PK, o `update_all` vai atualizar um registro só também :slightly_smiling_face: